core/integration/module_cli_test.go

```diff
@@ -1300,3 +1322,266 @@ func (f *Foo) ContainerEcho(ctx context.Context, input string) (string, error) {
+			},
+			verifySetup: daggerjsonContains([]string{`"github.com/shykes/daggerverse/docker@main"`}),
+			updateCmd:   []string{"update", "github.com/shykes/daggerverse/docker"},
+			verifyFns: []func(t *testctx.T, daggerjson string){
+				daggerjsonContains([]string{`"github.com/shykes/daggerverse/docker@main`}),
+				daggerjsonNotContains([]string{`"github.com/shykes/daggerverse/docker"`}),
+			},
+			expectedError: "",
+		},
+		{
+			name:    "existing dep have version, update cmd dont have version",
+			initCmd: []string{"init", "--sdk=go", "--name=foo", "--source=."},
+			installCmds: [][]string{
+				{"install", "github.com/shykes/daggerverse/docker@v0.4.2"},
+			},
+			verifySetup: daggerjsonContains([]string{`"github.com/shykes/daggerverse/docker@docker/v0.4.2"`}),
+			updateCmd:   []string{"update", "github.com/shykes/daggerverse/docker"},
+			verifyFns: []func(t *testctx.T, daggerjson string){
+				daggerjsonContains([]string{`github.com/shykes/daggerverse/docker@docker/v0.4.2`}),
+				daggerjsonNotContains([]string{`"github.com/shykes/daggerverse/docker"`}),
+			},
+			expectedError: "",
+		},
+		{
+			name:    "existing dep dont have version, update cmd use name without version",
+			initCmd: []string{"init", "--sdk=go", "--name=foo", "--source=."},
+			installCmds: [][]string{
+				{"install", "github.com/shykes/daggerverse/docker"},
+			},
+			verifySetup: daggerjsonContains([]string{`"github.com/shykes/daggerverse/docker"`}),
+			updateCmd:   []string{"update", "docker"},
+			verifyFns: []func(t *testctx.T, daggerjson string){
+				daggerjsonContains([]string{`"github.com/shykes/daggerverse/docker"`}),
+			},
+			expectedError: "",
+		},
+		{
+			name:    "existing dep use branch, update cmd use name without version",
+			initCmd: []string{"init", "--sdk=go", "--name=foo", "--source=."},
+			installCmds: [][]string{
+				{"install", "github.com/shykes/daggerverse/docker@main"},
+			},
+			verifySetup: daggerjsonContains([]string{`"github.com/shykes/daggerverse/docker@main"`}),
+			updateCmd:   []string{"update", "docker"},
+			verifyFns: []func(t *testctx.T, daggerjson string){
+				daggerjsonContains([]string{`"github.com/shykes/daggerverse/docker@main`}),
+				daggerjsonNotContains([]string{`"github.com/shykes/daggerverse/docker"`}),
+			},
+			expectedError: "",
+		},
+		{
+			name:    "existing dep have version, update cmd use name without version",
+			initCmd: []string{"init", "--sdk=go", "--name=foo", "--source=."},
+			installCmds: [][]string{
+				{"install", "github.com/shykes/daggerverse/docker@v0.4.2"},
+			},
+			verifySetup: daggerjsonContains([]string{`"github.com/shykes/daggerverse/docker@docker/v0.4.2"`}),
+			updateCmd:   []string{"update", "docker"},
+			verifyFns: []func(t *testctx.T, daggerjson string){
+				daggerjsonContains([]string{`github.com/shykes/daggerverse/docker@docker/v0.4.2`}),
+				daggerjsonNotContains([]string{`"github.com/shykes/daggerverse/docker"`}),
+			},
+			expectedError: "",
+		},
```

Comment:
This test (and maybe some others) need to check that the `pin` value is updated. You can do this by manually setting the `pin` in `dagger.json` to an old commit for this repo, and then checking it's correctly modified to the actual v0.4.2 commit.

Reply:
this is a great suggestion. I refactored tests to use old commit to start with and then verify that the commit is moved to the actual commit.

I also made changes to use creating `dagger.json` manually rather than using `dagger install` to avoid manipulating the `dagger.json` file.